impl/tests/scripts/test04.sh

```diff
@@ -0,0 +1,46 @@
+../common/start_test.sh $1 $2
+cd ../mountpoint
+
+mkdir -p a/aa/aaa/aaaa/aaaaa
+cd a
+cd aa
+cd aaa
+cd ../../..
+ls > files.txt
+md5sum files.txt >> ../$2
+touch a/aa/aaa/aaaa/aaaaa/blah.txt
+ls > files.txt
+md5sum files.txt >> ../$2
+cd a/aa/aaa/aaaa/aaaaa/
+ls > files.txt
+md5sum files.txt >> ../$2
+cd ..
+cd ..
+cd ..
+cd ..
+cd ..
+ls > files.txt
+md5sum files.txt >> ../$2
+mkdir -p b/bb/bbb/bbbb/bbbbb
+cp a/aa/aaa/aaaa/aaaaa/blah.txt b/bb/bbb/bbbb/bbbbb/try.txt
+ls > files.txt
+md5sum files.txt >> ../$2
+md5sum a/aa/aaa/aaaa/aaaaa/* >> ../$2
+md5sum b/bb/bbb/bbbb/bbbbb/* >> ../$2
+
+mv a c
+ls > files.txt
+md5sum files.txt >> ../$2
+#md5sum c >> ../$2
+#md5sum b >> ../$2
+
+rm -rf c
+rm -rf b
+
+ls > files.txt
+md5sum files.txt >> ../$2
+rm *
+
+cd ..
+
+./common/end_test.sh
```

Comment:
Structuring the code to have functions, and not just statements is probably a better idea, and also conveys what the function does. Agreed, that the `start_test.sh` and `end_test.sh` do convey what they are doing, but functions are better.